src/main/java/com/atlauncher/App.java

```diff
@@ -165,7 +169,60 @@ public void run() {
         }
 
         LogManager.info("ATLauncher Version: " + Constants.VERSION);
-        LogManager.info("Operating System: " + System.getProperty("os.name"));
+        if(System.getProperty("os.name").equals("Linux"))
+        {
+            String distro = "Unknown Distribution";
+            if(new File("/etc/lsb-release").exists())
+            {
+                try
+                {
+                    String line = "";
+                    BufferedReader lsbReader = new BufferedReader(new FileReader(new File("/etc/lsb-release")));
+                    while((line = lsbReader.readLine()) != null) {
+                        if(line.indexOf("DISTRIB_DESCRIPTION") != -1) {
+                            distro = line.substring(20).replace("\"","");
+                        }
+                    }
+                }
+                catch(IOException e) {
+                    LogManager.warn("Unable to read lsb-release:" + e.getMessage());
+                }
+            }
+            else if(new File("/etc/os-release").exists())
+            {
+                try
+                {
+                    String line = "";
+                    BufferedReader osReader = new BufferedReader(new FileReader(new File("/etc/os-release")));
+                    while((line = osReader.readLine()) != null) {
+                        if(line.indexOf("PRETTY_NAME") != -1) {
+                            distro = line.substring(12).replace("\"","");
+                        }
+                    }
+                }
+                catch(IOException e) {
+                    LogManager.warn("Unable to read os-release:" + e.getMessage());
+                }
+            }
+            else if(new File("/etc/redhat-release").exists())
```

Comment:
It gets the OS but that will just state Linux, Linux is the operating
system but the point of this code is finding out what Linux Distribution is
in use and if possible the release of the distribution.
On 14 Jan 2015 16:36, "Jamie" <notifications@github.com> wrote:

> In src/main/java/com/atlauncher/App.java
> <https://github.com/ATLauncher/ATLauncher/pull/126#discussion-diff-22949096>
> :
>
> > +            {
> > +                try
> > +                {
> > +                    String line = "";
> > +                    BufferedReader osReader = new BufferedReader(new FileReader(new File("/etc/os-release")));
> > +                    while((line = osReader.readLine()) != null) {
> > +                        if(line.indexOf("PRETTY_NAME") != -1) {
> > +                            distro = line.substring(12).replace("\"","");
> > +                        }
> > +                    }
> > +                }
> > +                catch(IOException e) {
> > +                    LogManager.warn("Unable to read os-release:" + e.getMessage());
> > +                }
> > +            }
> > +            else if(new File("/etc/redhat-release").exists())
>
> Can't you just use System.getProperty("os.name"); to get the os
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ATLauncher/ATLauncher/pull/126/files#r22949096>.
>